skyportal/handlers/api/taxonomy.py

```diff
@@ -0,0 +1,206 @@
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            comment_id:
+                              type: integer
+                              description: New taxonomy ID
+        """
+        data = self.get_json()
+        name = data.get('name', None)
+        if name is None:
+            return self.error(f"A name must be provided for a taxonomy")
+
+        version = data.get('version', None)
+        if version is None:
+            return self.error(
+                f"A version string must be provided for a taxonomy"
+            )
+
+        others = (
+            Taxonomy.query
+            .filter(Taxonomy.name == name)
+            .filter(Taxonomy.version == version)
+            .all()
+        )
+
+        if len(others) != 0:
+            return self.error(
+                f"That version/name combination is already "
+                "present. If you really want to replace this "
+                "then delete the appropriate entry."
+            )
+
+        # Ensure a valid taxonomy
+        hierarchy = data.get('hierarchy', None)
+        if hierarchy is None:
+            return self.error(f"A JSON of the taxonomy must be given")
+
+        try:
+            validate(hierarchy, schema)
+        except JSONValidationError:
+            return self.error(
+                f"Hierarchy does not validate against the schema."
+            )
+
+        # establish the groups to use
+        user_group_ids = [g.id for g in self.current_user.groups]
+        group_ids = data.pop("group_ids", user_group_ids)
+        if group_ids == []:
+            group_ids = user_group_ids
+        group_ids = [gid for gid in group_ids if gid in user_group_ids]
+        if not group_ids:
+            return self.error(f"Invalid group IDs field ({group_ids}): "
+                              "You must provide one or more valid group IDs.")
+        groups = Group.query.filter(Group.id.in_(group_ids)).all()
+
+        provenance = data.get('provenance', None)
+
+        # update others with this name
+        # TODO: deal with the same name but different groups?
+        isLatest = data.get('isLatest', True)
+        if isLatest:
+            DBSession().query(Taxonomy) \
+                       .filter(Taxonomy.name == name) \
+                       .update({'isLatest': False})
+
+        taxonomy = Taxonomy(
+            name=name,
+            hierarchy=hierarchy,
+            provenance=provenance,
+            version=version,
+            isLatest=isLatest,
+            groups=groups
+        )
+
+        DBSession().add(taxonomy)
+        DBSession().commit()
+
+        return self.success(data={'taxonomy_id': taxonomy.id})
+
+    @permissions(['Taxonomy Delete'])
```

Comment:
Why have a separate ACL for post/delete?

Reply:
I REALLY want `Taxonomy` to be append-only for the vast majority of the uses of SkyPortal. Since we need to keep track of the classifications from older taxonomies. Having a separate ACL for `Taxonomy Delete` ensures that a role needs to get explicit permissions to delete a `Taxonomy`. For now `Super Admin` is the only role that gets this ACL by default.